src/app/components/common/node-editor/custom-notification-node/custom-notification-node.component.ts

```diff
@@ -0,0 +1,161 @@
+import { isEmpty } from 'lodash';
+import { Component, Input, HostBinding, ChangeDetectorRef, OnChanges, ElementRef } from "@angular/core";
+import { ClassicPreset } from "rete";
+import { KeyValue } from "@angular/common";
+import { ActivatedRoute, NavigationEnd, Router } from "@angular/router";
+import { RolesService, ConfigurationService } from "../../../../services";
+import { DocService } from "../../../../services/doc.service";
+import { FlowEditorService } from "../flow-editor.service";
+import { Subject, Subscription } from "rxjs";
+
+@Component({
+  selector: 'app-custom-notification-node',
+  templateUrl: './custom-notification-node.component.html',
+  styleUrls: ['./custom-notification-node.component.css'],
+  host: {
+    "data-testid": "notification-node"
+  }
+})
+export class CustomNotificationNodeComponent implements OnChanges {
+
+  @Input() data!: ClassicPreset.Node;
+  @Input() emit!: (data: any) => void;
+  @Input() rendered!: () => void;
+
+  seed = 0;
+  helpText = '';
+  showPluginName = false;
+  
+  notification = {
+    name: "",
+    channel: "",
+    enable: false,
+    notificationType: "",
+    retriggerTime: "",
+    rule: "",
+    isDeliveryEnabled: false,
+    isServiceEnabled: false
+  }
+
+  isNotificationNode: boolean = false;
+
+  subscription: Subscription;
+  pluginName = '';
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  
+  showPlusIcon = false;
+  showDeleteIcon = false;
+  nodeId = '';
+
+  @HostBinding("class.selected") get selected() {
+    return this.data.selected;
+  }
+
+  constructor(private cdr: ChangeDetectorRef,   
+    private docService: DocService,
+    private router: Router,
+    private route: ActivatedRoute,
+    public flowEditorService: FlowEditorService,
+    public rolesService: RolesService,
+    public configurationService: ConfigurationService,
+    private elRef: ElementRef) {
+    this.router.routeReuseStrategy.shouldReuseRoute = function () {
+      return false;
+    };
+    this.router.navigated = false;
+    this.subscription = this.router.events.subscribe((event) => {
+      if (event instanceof NavigationEnd) {
+        this.router.navigated = false;
+      }
+    });
+  }
+
+  ngOnChanges(): void {
+    this.nodeId = this.data.id;
+    if (this.data.label === 'Notification') {
+      this.elRef.nativeElement.style.borderColor = "#0E9BD8";
+      this.isNotificationNode = true;    
+      if ('notifications' == 'notifications') {
+        if (!isEmpty(this.data.controls)) {
+          this.notification.name = this.data.controls.nameControl['name'];
```

Comment:
Better to use data.controls directly in html instead of creating separate object and initialise its properties with incoming data

Reply:
We are passing this notification data to `flowEditorService.showItemsInQuickview` and it is of Notification type i.e.-


```
 {
    name: string
    rule: string
    channel: number
    notificationType: string
    retriggerTime: number
    enable: boolean
  }
```

If I pass it as it is i.e. `this.data.controls` then I'll have to access all controls as `this.notification.nameControl['name']` etc. in node editor file and it can't be typecast as `Notification`. So I think it is better to use an notification object here.